### PR TITLE
funds-manager: Add endpoint to fetch token balances for hot wallets

### PR DESCRIPTION
**Pull request by joeykraut**
### Purpose
This PR adds an endpoint to query the token balances for hot wallets on-chain. The endpoint fetches all hot wallets and takes the desired mints as query parameters and looks them up on their ERC20 contracts.

### Testing
- Tested the endpoint locally against a hot wallet I funded with a dummy erc20

### CODE REVIEW

File: funds-manager/funds-manager-server/src/handlers.rs [hunk 1]
```diff
@@ -67,8 +68,8 @@ pub(crate) async fn get_deposit_address_handler(
     query_params: HashMap<String, String>,
     server: Arc<Server>,
 ) -> Result<Json, warp::Rejection> {
-    let mint = query_params.get(MINT_QUERY_PARAM).ok_or_else(|| {
-        warp::reject::custom(ApiError::BadRequest("Missing 'mint' query parameter".to_string()))
+    let mint = query_params.get(MINTS_QUERY_PARAM).ok_or_else(|| {
+        warp::reject::custom(ApiError::BadRequest("Missing 'mints' query parameter".to_string()))
```
Comment (akirillo) [inline funds-manager/funds-manager-server/src/handlers.rs:72]: pedantic nit: I would keep this parameter as the singular `mint` to make the API more readable (at the expense of an extra constant)

File: funds-manager/funds-manager-server/src/main.rs [hunk 2]
```diff
@@ -342,13 +342,19 @@ async fn main() -> Result<(), Box<dyn Error>> {
...
-        .and(warp::path(CREATE_HOT_WALLET_ROUTE))
         .and(with_hmac_auth(server.clone()))
         .map(with_json_body::<CreateHotWalletRequest>)
         .and_then(identity)
         .and(with_server(server.clone()))
         .and_then(create_hot_wallet_handler);
 
+    let get_hot_wallet_balances = warp::get()
+        .and(warp::path("hot-wallets"))
+        .and(with_hmac_auth(server.clone()))
```
Comment (akirillo) [inline funds-manager/funds-manager-server/src/main.rs:353]: better to be on the safe side, but does this endpoint need hmac auth? i thought the idea was just to protect the "write" / post endpoints

**Review by akirillo — APPROVED** [review]

File: funds-manager/funds-manager-server/src/main.rs (hunk 2 above)
Comment (joeykraut) [reply funds-manager/funds-manager-server/src/main.rs:353]: without hmac auth anyone would be able to view our balances, that's certainly not okay

File: funds-manager/funds-manager-server/src/handlers.rs (hunk 1 above)
Comment (joeykraut) [reply funds-manager/funds-manager-server/src/handlers.rs:72]: disagree that it's more readable, i'll keep it this way. this is a very common pattern to accept list arguments

**Pull request merged**